ns3_adapter/src/ns3_client.cpp

```diff
@@ -13,86 +14,134 @@ NS3Client::~NS3Client() {
     }catch(...){}
 }
 
-bool NS3Client::registermsg(const std::shared_ptr<std::vector<uint8_t>>&message, const std::string &remote_address, unsigned short remote_port,
-                            unsigned short local_port)
-{
-    bool success = false;
-    
-    if (connect(remote_address, remote_port, local_port))
+
+bool NS3Client::connect_registration(const std::string &remote_address, unsigned short remote_port) {
+    boost::system::error_code ignored_ec;
+     //If we are already connected return false
+    if(running_) return false;
+    //Get remote endpoint
+    try
     {
-        bool send_success = sendNS3Message(message);
-        if (send_success)
-        { 
-            // ROS_DEBUG_STREAM("Handshake Message sent successfully");
-            success = true;
-        }
-        // else ROS_DEBUG_STREAM("Handshake Message send failed");
+        remote_udp_ep_registration_ = boost::asio::ip::udp::endpoint(boost::asio::ip::address::from_string(remote_address),remote_port);
+    }catch(std::exception e)
+    {
+        ignored_ec = boost::asio::error::invalid_argument;
```

Comment:
Why are we catching an exception and just rethrowing the same exception. Either we should log something, or add information and throw a more specific exception. Also what is the reason we are setting this ignored_ec (I assume error code). We never do anything with this error code.

Reply:
removed this function